src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -792,11 +794,21 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
             this.storeLastMembershipFingerprints();
 
             if (anyLeft) {
-                logger.debug(`Member(s) have left: queueing sender key rotation`);
-                this.makeNewKeyTimeout = setTimeout(this.onRotateKeyTimeout, MAKE_KEY_DELAY);
+                if (newMembershipIds.size === 1) {
+                    logger.debug(`New member(s) have left: doing immediate sender key rotation`);
+                    this.doRotateKey(false);
+                } else {
+                    logger.debug(`New member(s) have left: queueing sender key rotation`);
+                    this.makeNewKeyTimeout = setTimeout(() => this.doRotateKey(true), MAKE_KEY_DELAY);
+                }
```

Comment:
It would be useful to have a comment around this area about why one would/wouldn't want there to be a delay on key rotation

Reply:
So, I went to add the comments and I found myself reworking the logic and catering for more scenarios. It will need a re-review.